docs/pages/enroll-resources/kubernetes-access/kubernetes-access-examples.mdx

```diff
@@ -0,0 +1,139 @@
+to specific namespaces, enforcing read-only permissions, or restricting administrative actions. 
+
+<Admonition type="tip">
+Reference [Setting Up Teleport Access Controls for Kubernetes](./manage-access.mdx) before proceeding, if you have not done so already.
+</Admonition>
+
+Here are a few common use cases. These are sample examples to get you started; for more in-depth coverage of roles in Kubernetes access, please visit the 
+[Access Controls guide](./controls.mdx) for Kubernetes.
+
+## Use case 1: Example role for developers
+
+In many organizations, developers need limited access to Kubernetes clusters, typically restricted to their development environments. 
+
+Below is an example of a role that provides developers with access to only the dev namespace:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: dev-k8s-access
+spec:
+  allow:
+    kubernetes_groups: ["dev-group"]
+    kubernetes_labels:
+      "environment": "dev"  # Allow access only to development clusters
+    kubernetes_resources:
+      - kind: pod
+        namespace: dev
+      - kind: service
+        namespace: dev
+  deny:
+    kubernetes_labels:
+      "environment": "prod"  # Deny access to production clusters
+```
+
+In this example, the developers in the group dev-group are allowed to access the `dev` namespace's pods and services. Access to any production environments is explicitly denied, using the `deny` directive.
+
+## Use case 2: Read-only access for auditors
+
+In some scenarios, you may need to provide auditors with read-only access to a Kubernetes cluster for compliance purposes. The following auditor role grants read-only access to all namespaces 
+but does not allow any write operations:
+
+```code
+kind: role
+version: v6
+metadata:
+  name: auditor-k8s-read-only
+spec:
+  allow:
+    kubernetes_groups: ["auditor-group"]
+    kubernetes_labels:
+      "*": "*"  # Access to all clusters
+    kubernetes_resources:
+      - kind: pod
+        verbs: ["get", "list", "watch"]
+      - kind: service
+        verbs: ["get", "list", "watch"]
+  deny:
+    kubernetes_resources:
+      - kind: pod
+        verbs: ["create", "update", "delete"]  # Deny any modification operations
+      - kind: service
+        verbs: ["create", "update", "delete"]
+```
+
+In this example, the auditor-group is allowed to access all namespaces (*), but only with `get`, `list`, and `watch` permissions (read-only). Modifications such as creating, updating, or deleting resources are denied.
+
+## Use case 3: Admin access to all Kubernetes clusters
+
+In organizations with DevOps or platform engineering teams, certain users need full administrative access to all Kubernetes clusters for management purposes. This role grants full access to all Kubernetes resources across all clusters, allowing the user to perform any action on any resource.
+
+```code
+kind: role
+version: v6
+metadata:
+  name: admin-k8s-access
```

Comment:
might want to update this to role v7 now https://goteleport.com/docs/enroll-resources/kubernetes-access/manage-access/#define-a-role